### DISCUSSION TITLE
Javers retrieve the ListChange objects (It is giving only reference id for the objects and not actual object values)

### DISCUSSION THREAD

**Discussion by Anupam2528** [javers/javers#1235] [Q&A]
Using Javers for the fetching historical data (Audit purpose) in Java Springboot. I have a entity which is having one to many relationship with other child entities. Below is the example what we are doing. Example :

@Entity
@Getter
@Setter
@NoArgsConstructor
class ParentEntity{

@Id
@GenericGenerator(name = "UUIDGenerator", strategy = "uuid2")
@GeneratedValue(generator = "UUIDGenerator")
@Column(name = "id", updatable = false, nullable = false)
private UUID id;

@Column(name = "customer_id")
private String customerId;

@OneToMany(mappedBy = "parentEntity", cascade = CascadeType.ALL, orphanRemoval = true)
private List<ChildEntity> childEntity;

} 
@Entity
@Getter
@Setter
class ChildEntity{

@Id
@GenericGenerator(name = "UUIDGenerator", strategy = "uuid2")
@GeneratedValue(generator = "UUIDGenerator")
@Column(name = "id", updatable = false, nullable = false)
protected UUID id;

@ManyToOne(fetch = FetchType.LAZY)
private ParentEntity parentEntity;

}
If I made a change for the object of Child entity which is present in Parent, we are getting the ListChange object but it does not contains the actual object changed, it is having the reference global id inside it.

We are using below query :

javers.findChanges(QueryBuilder.byInstanceId(id, classname).withSnapshotTypeUpdate().build());
Is there any way to fetch the exact object values from the Javers tables?

I am getting the response which has the global id for respective updated object in Left and Right.

I am looking for the objects which got changed (Previous version and updated version)

**Comment by pravinvignesh** [comment]
Did anyone know solution for the above query? How can i fetch the values of object instead of reference inorder to display it in UI @bartoszwalacik

**Comment by bartoszwalacik** [comment]
The answer for this question can be easily found in the docs, see https://javers.org/documentation/jql-examples/#find-methods
